Tribler/community/popular/community.py

```diff
@@ -0,0 +1,501 @@
+
+        if peer:
+            self.broadcast_message(packet, peer=peer)
+        else:
+            for connected_peer in self.get_peers():
+                self.broadcast_message(packet, peer=connected_peer)
+
+    def send_content_info_response(self, peer, identifier, content_type, response_list):
+        num_results = len(response_list)
+        current_index = 0
+        page_num = 1
+        while current_index < num_results:
+            serialized_results, current_index, page_size = self.pack_sized(response_list, MAX_PACKET_PAYLOAD_SIZE,
+                                                                           start_index=current_index)
+            if not serialized_results:
+                self.logger.info("Item too big probably to fit into package. Skipping it")
+                current_index += 1
+            else:
+                pagination = Pagination(page_num, page_size, num_results, more=current_index == num_results)
+                response_payload = ContentInfoResponse(identifier, content_type, serialized_results, pagination)
+                packet = self.create_message_packet(MSG_CONTENT_INFO_RESPONSE, response_payload)
+                self.broadcast_message(packet, peer=peer)
+
+    def send_torrent_search_request(self, query):
+        self.send_content_info_request(SEARCH_TORRENT_REQUEST, query)
+
+    def send_channel_search_request(self, query):
+        raise NotImplementedError("Not implemented yet. Waiting for all channel 2.0")
+
+    # CONTENT REPOSITORY STUFFS
+
+    def publish_next_content(self):
+        """
+        Publishes the next content from the queue to the subscribers.
+        Does nothing if there are none subscribers.
+        Only Torrent health response is published at the moment.
+        """
+        self.logger.info("Content to publish: %d", self.content_repository.num_content())
+        if not self.subscribers:
+            self.logger.info("No subscribers found. Not publishing anything")
+            return
+
+        content_type, content = self.content_repository.pop_content()
+        if content_type is None:
+            self.logger.error(ERROR_NO_CONTENT)
+            return
+
+        self.logger.info("Publishing content[type:%d]", content_type)
+        if content_type == TYPE_TORRENT_HEALTH:
+            infohash, seeders, leechers, timestamp = content
+            payload = TorrentHealthPayload(infohash, seeders, leechers, timestamp)
+            self.send_torrent_health_response(payload)
+
+    def publish_latest_torrents(self, peer):
+        """
+        Publishes the latest torrents in local database to the given peer.
+        """
+        torrents = self.content_repository.get_top_torrents()
+        self.logger.info("Publishing %d torrents to peer %s", len(torrents), peer)
+        for torrent in torrents:
+            infohash, seeders, leechers, timestamp = torrent[:4]
+            payload = TorrentHealthPayload(infohash, seeders, leechers, timestamp)
```

Comment:
This is more an IPv8 comment: I think the community designs are diverging and every developer maintains its own style of defining messages. @qstokkink do you agree?